source/common/http/header_map_impl.cc

```diff
@@ -182,8 +192,13 @@ void HeaderString::setInteger(uint64_t value) {
   }
 
   case Type::Inline:
+    // buffer_.dynamic_ should always point at inline_buffer_ for Type::Inline.
+    ASSERT(buffer_.dynamic_ == inline_buffer_);
   case Type::Dynamic: {
     // Whether dynamic or inline the buffer is guaranteed to be large enough.
+    ASSERT(dynamic_capacity_ >= MaxIntegerLength);
+    // It's safe to use buffer.dynamic_, since buffer.ref_ is union aliased.
+    ASSERT(&buffer_.dynamic_ == &buffer_.ref_);
```

Comment:
When would this ever not be true? Seems like we are testing the compiler here, but maybe I'm missing something.

Reply:
If you made a change to the code to change a union to struct, for example :)

Reply:
It would be nice if that could be a `static_assert`, then. Obviously not possible in its current form. Can you take advantage of something like `typeid` ?